src/back/spv/block.rs

```diff
@@ -1955,19 +2012,31 @@ impl<'w> BlockContext<'w> {
             }
         }
 
-        let termination = match exit_id {
-            Some(id) => Instruction::branch(id),
+        let termination = match exit {
             // This can happen if the last branch had all the paths
             // leading out of the graph (i.e. returning).
             // Or it may be the end of the self.function.
-            None => match self.ir_function.result {
+            BlockExit::Return => match self.ir_function.result {
                 Some(ref result) if self.function.entry_point_context.is_none() => {
                     let type_id = self.get_type_id(LookupType::Handle(result.ty));
                     let null_id = self.writer.write_constant_null(type_id);
                     Instruction::return_value(null_id)
                 }
                 _ => Instruction::return_void(),
             },
+            BlockExit::Branch { block_id } => Instruction::branch(block_id),
+            BlockExit::BreakIf {
+                condition,
+                preamble_id,
+            } => {
+                let condition_id = self.cached[condition];
+
+                Instruction::branch_conditional(
+                    condition_id,
+                    loop_context.break_id.unwrap(),
```

Comment:
This `unwrap` makes me wonder whether `LoopContext` and `BlockExit` couldn't be fused into a single enum. But that can be another PR.

Reply:
They could be merged but it would make it more painful to handle them in the `Break` and `Continue` statements